zeus/device/cpu/rapl.py

```diff
@@ -261,6 +265,65 @@ def supportsGetDramEnergyConsumption(self) -> bool:
         return self.dram is not None
 
 
+class ZeusdRAPLCPU(RAPLCPU):
+    """A RAPLCPU that interfaces with RAPL via zeusd.
+
+    The parent RAPLCPU class requires root privileges to interface with RAPL.
+    ZeusdRAPLCPU (this class) overrides RAPLCPU's methods so that they instead send requests
+    to the Zeus daemon, which will interface with RAPL on behalf of ZeusdRAPLCPU. As a result,
+    ZeusdRAPLCPU does not need root privileges to monitor CPU and DRAM energy consumption.
+
+    See [here](https://ml.energy/zeus/getting_started/#system-privileges)
+    for details on system privileges required.
+    """
+
+    def __init__(
+        self,
+        cpu_index: int,
+        zeusd_sock_path: str = "/var/run/zeusd.sock",
+    ) -> None:
+        """Initialize the Intel CPU with a specified index."""
+        self.cpu_index = cpu_index
+
+        self._client = httpx.Client(transport=httpx.HTTPTransport(uds=zeusd_sock_path))
+        self._url_prefix = f"http://zeusd/cpu/{cpu_index}"
+
+    def getTotalEnergyConsumption(self) -> CpuDramMeasurement:
+        """Returns the total energy consumption of the specified powerzone. Units: mJ."""
+        resp = self._client.post(
+            self._url_prefix + "/get_index_energy",
+            json={
+                "cpu": True,
+                "dram": True,
+            },
+        )
+        if resp.status_code != 200:
+            raise ZeusdError(f"Failed to get total energy consumption: {resp.text}")
+
+        data = resp.json()
+        cpu_uj = data.get("cpu_energy_uj")
+        dram_uj = data.get("dram_energy_uj")
+        cpu_mj = None if cpu_uj is None else cpu_uj / 1000
+        dram_mj = None if dram_uj is None else dram_uj / 1000
+
+        return CpuDramMeasurement(cpu_mj=cpu_mj, dram_mj=dram_mj)
+
+    def supportsGetDramEnergyConsumption(self) -> bool:
+        """Returns True if the specified CPU powerzone supports retrieving the subpackage energy consumption."""
+        resp = self._client.get(
+            self._url_prefix + "/supportsDramEnergy",
```

Comment:
All other routes use camel case for the path. Please make this camel case as well.